spanner-jdbc: Unable to update JDBC driver to depend on latest spanner version

PR #28 is failing builds with the following errors:

```
[ERROR] COMPILATION ERROR :
[INFO] -------------------------------------------------------------
[ERROR] /tmpfs/src/github/java-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcAbortedTransactionTest.java:[126,16] cannot access com.google.spanner.v1.SpannerGrpc
  class file for com.google.spanner.v1.SpannerGrpc not found
[ERROR] /tmpfs/src/github/java-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcAbortedTransactionTest.java:[134,13] no suitable method found for addService(com.google.cloud.spanner.MockSpannerServiceImpl)
    method io.grpc.ServerBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.ServerBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
[ERROR] /tmpfs/src/github/java-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcDriverTest.java:[60,52] no suitable method found for addService(com.google.cloud.spanner.MockSpannerServiceImpl)
    method io.grpc.ServerBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.ServerBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
[ERROR] /tmpfs/src/github/java-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/AbortedTest.java:[101,13] no suitable method found for addService(com.google.cloud.spanner.MockSpannerServiceImpl)
    method io.grpc.ServerBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.ServerBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
[ERROR] /tmpfs/src/github/java-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcDatabaseMetaDataWithMockedServerTest.java:[76,52] no suitable method found for addService(com.google.cloud.spanner.MockSpannerServiceImpl)
    method io.grpc.ServerBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.ServerBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
[ERROR] /tmpfs/src/github/java-spanner-jdbc/src/test/java/com/google/cloud/spanner/jdbc/JdbcGrpcErrorTest.java:[110,13] no suitable method found for addService(com.google.cloud.spanner.MockSpannerServiceImpl)
    method io.grpc.ServerBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.ServerBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.ServerServiceDefinition) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.ServerServiceDefinition)
    method io.grpc.internal.AbstractServerImplBuilder.addService(io.grpc.BindableService) is not applicable
      (argument mismatch; com.google.cloud.spanner.MockSpannerServiceImpl cannot be converted to io.grpc.BindableService)
```

@olavloite could these tests be fixed or is there something that we need to fix in the Spanner client?

--- a/pom.xml
+++ b/pom.xml
@@ -69,7 +69,7 @@
     <protobuf.version>3.11.3</protobuf.version>
     <threeten.version>1.4.1</threeten.version>
     <google.api-common.version>1.8.1</google.api-common.version>
-    <spanner.version>1.47.0</spanner.version>
+    <spanner.version>1.49.2</spanner.version>
     <hamcrest.version>2.2</hamcrest.version>
     <error-prone.version>2.3.4</error-prone.version>
     <google.auth.version>0.20.0</google.auth.version>
@@ -242,6 +242,24 @@
       <artifactId>google-cloud-spanner</artifactId>
       <version>${spanner.version}</version>
       <type>test-jar</type>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>grpc-google-cloud-spanner-v1</artifactId>
+      <version>${spanner.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>grpc-google-cloud-spanner-admin-instance-v1</artifactId>
+      <version>${spanner.version}</version>
+      <scope>test</scope>
+    </dependency>
+    <dependency>
+      <groupId>com.google.api.grpc</groupId>
+      <artifactId>grpc-google-cloud-spanner-admin-database-v1</artifactId>
+      <version>${spanner.version}</version>
       <scope>test</scope>
     </dependency>
     <dependency>

--- a/pom.xml
+++ b/pom.xml
@@ -339,7 +339,12 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-dependency-plugin</artifactId>
           <configuration>
-            <ignoredUnusedDeclaredDependencies>com.google.api:gax-grpc</ignoredUnusedDeclaredDependencies>
+            <ignoredUnusedDeclaredDependencies>
+              <ignoredDependency>com.google.api:gax-grpc</ignoredDependency>
+              <ignoredDependency>com.google.api.grpc:grpc-google-cloud-spanner-v1</ignoredDependency>
+              <ignoredDependency>com.google.api.grpc:grpc-google-cloud-spanner-admin-instance-v1</ignoredDependency>
+              <ignoredDependency>com.google.api.grpc:grpc-google-cloud-spanner-admin-database-v1</ignoredDependency>
+            </ignoredUnusedDeclaredDependencies>
           </configuration>
         </plugin>
       </plugins>

